### PR TITLE
Fix for inactive cells in updates

### LINKED ISSUE #7256: Inactive cells are maybe given to the update algorithm
Originally these values were not provided to the update algorithm, but this might have changed. We should validate that inactive parameters are not provided to the update algorithm.

### PR DESCRIPTION
**Pull request by frode-aarstad**
**Issue**
Resolves #7256 

**Approach**
_Short description of the approach_

(Screenshot of new behavior in GUI if applicable)

- [ ] PR title captures the intent of the changes, and is fitting for release notes.
- [ ] Added appropriate release note label
- [ ] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [ ] Make sure tests pass locally (after every commit!)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Create Backport PR to latest release

### CODE REVIEW

File: tests/integration_tests/storage/test_field_parameter.py [hunk 1]
```diff
@@ -178,7 +185,7 @@ def test_parameter_update_with_inactive_cells_xtgeo_grdecl(tmpdir):
 import os
 if __name__ == "__main__":
     if not os.path.exists("my_param.grdecl"):
-        values = np.random.standard_normal(4*4)
+        values = np.random.standard_normal(123*111*6)
```
Comment (dafeda) [inline tests/integration_tests/storage/test_field_parameter.py:188]:
```suggestion
        values = np.random.standard_normal(NCOL*NROW*NLAY)
```

**Review by dafeda — APPROVED** [review]
Nice 👍

File: tests/integration_tests/storage/test_field_parameter.py (hunk 1 above)
Comment (frode-aarstad) [reply tests/integration_tests/storage/test_field_parameter.py]: Fixed this but needed a slightly different approach

**Review by oyvindeide — CHANGES_REQUESTED** [review]

File: src/ert/analysis/_es_update.py [hunk 2]
```diff
@@ -252,7 +252,18 @@ def _save_temp_storage_to_disk(
                     }
                 )
                 target_fs.save_parameters(key, realization, dataset)
-            elif isinstance(config_node, (Field, SurfaceConfig)):
+            elif isinstance(config_node, Field):
+                assert isinstance(matrix, np.ndarray)
+                ma = np.ma.MaskedArray(  # type: ignore
```
[the remark this answers is not in the document]
Comment (frode-aarstad) [reply src/ert/analysis/_es_update.py:257]: np.nan 

https://gitkraken.dev/link/dnNjb2RlOi8vZWFtb2Rpby5naXRsZW5zL2xpbmsvci9hMWRlYjdmN2FhODhkYjMwZmMwZjA5MGZmN2JkMDQ4YWU5YTUzYWI2L2Yvc3JjL2VydC9maWVsZF91dGlscy9ncmRlY2xfaW8ucHk%2FdXJsPWdpdCU0MGdpdGh1Yi5jb20lM0FlcXVpbm9yJTJGZXJ0LmdpdCZsaW5lcz0yMzE%3D?origin=gitlens

File: src/ert/analysis/_es_update.py [hunk 3]
```diff
@@ -284,7 +295,15 @@ def _create_temporary_parameter_storage(
...
-        matrix = source_fs.load_parameters(param_group, iens_active_index)["values"]
+        ds = source_fs.load_parameters(param_group, iens_active_index)
+        ensemble_size = len(ds.realizations)
+        da = xr.DataArray(
+            [
+                np.ma.MaskedArray(data=d, mask=config_node.mask).compressed()  # type: ignore
+                for d in ds["values"].values.reshape(ensemble_size, -1)
+            ]
+        )
+        matrix = da.T.to_numpy()
```
[the remark this answers is not in the document]
Comment (frode-aarstad) [reply src/ert/analysis/_es_update.py:306]: It should be possible to add  `save_parameter()` to ParameterConfig(ABC) and implement that for Field, GenKw and Surface if that's what you are thinking of.

Perhaps its best to do in its own PR?

**Review by oyvindeide — APPROVED** [review]
LGTM!

**Pull request merged**